core/arch/arm/plat-sam/nsec-service/sm_platform_handler.c

```diff
@@ -10,13 +10,28 @@
 #include <kernel/tee_misc.h>
 #include <mm/core_memprot.h>
 #include <sam_sfr.h>
+#include <sam_pl310.h>
 #include <sm/optee_smc.h>
 #include <sm/sm.h>
 #include <smc_ids.h>
 
 static enum sm_handler_ret sam_sip_handler(struct thread_smc_args *args)
 {
 	switch (OPTEE_SMC_FUNC_NUM(args->a0)) {
+#ifdef CFG_PL310_SIP_PROTOCOL
```

Comment:
Do the accesses to the PL310 need to be serialized to protect against concurrency?

Reply:
Actually, in the Linux kernel, only PL310_ENABLED is called for only once. And it's assumed that the OS will keep operations serialized.

Reply:
Would it be OK with adding the following at the beginning of this function? Thanks.
```
/*
 * As all sama5 SoCs are single-core ones, check the code compiled for a
 * single core. No serializations done to protect against concurrency.
 */
COMPILE_TIME_ASSERT(CFG_TEE_CORE_NB_CORE == 1);
```

Reply:
Sounds good, please replace `COMPILE_TIME_ASSERT` with `static_assert`.

Reply:
Updated, thank you.